public/src/main/java/com/revenuecat/purchases/models/PurchaseOption.kt

```diff
@@ -1,36 +1,23 @@
 package com.revenuecat.purchases.models
 
 import android.os.Parcelable
-import com.revenuecat.purchases.parceler.JSONObjectParceler
-import kotlinx.parcelize.Parcelize
-import kotlinx.parcelize.TypeParceler
-import org.json.JSONObject
 
-@Parcelize
-@TypeParceler<JSONObject, JSONObjectParceler>()
-data class PurchaseOption(
+interface PurchaseOption : Parcelable {
     /**
      * Pricing phases defining a user's payment plan for the product over time.
      */
-    val pricingPhases: List<PricingPhase>,
+    val pricingPhases: List<PricingPhase>
 
     /**
-     * Tags defined on the base plan or offer in the Google dashboard.
+     * Tags defined on the base plan or offer. Empty for Amazon.
      */
-    val tags: List<String> = listOf(),
-
-    /**
-     * Token used to purchase. Only used for Google BC5 subscriptions, null otherwise.
-     */
-    val token: String? = null
-) : Parcelable {
+    val tags: List<String>
 
     /**
      * True if this PurchaseOption represents a Google subscription base plan (rather than an offer).
-     * Not applicable for Amazon or one-time purchases.
+     * Not applicable for Amazon or INAPP products.
```

Comment:
Wonder if we should make this nullable for these cases... Otherwise we probably want to make sure this is always false in those cases. For one time purchases, do we have a single pricing phase? If so this could be true if I understand it correctly

Reply:
the way it's currently set up, we have the oneTimePrice directly in the StoreProduct, so purchaseoptions would be empty altogether. i'll update this comment and make a note in the OTP ticket we have to reconsider.